lumicks/pylake/kymotracker/tests/test_io.py

```diff
@@ -93,3 +103,46 @@ def test_kymotrackgroup_io(
         count_field = [key for key in data.keys() if "counts" in key][0]
         for track1, cnt in zip(tracks, data[count_field]):
             np.testing.assert_allclose([sampling_outcome] * len(track1.coordinate_idx), cnt)
+
+
+@pytest.mark.parametrize(
+    "version, read_with_version",
+    [[0, False], [1, False], [2, True]],
+)
+def test_csv_version(version, read_with_version):
+    # Test that header is parsed properly on CSV import
+    # Version 2 has 2 header lines, <2 only has 1 header line
+
+    track_coordinates = [
+        ((1, 2, 3), (2, 3, 4)),
+        ((2, 3, 4), (3, 4, 5)),
+    ]
+
+    test_data = np.zeros((8, 8))
+    for time_idx, position_idx in track_coordinates:
+        test_data[np.array(position_idx).astype(int), np.array(time_idx).astype(int)] = 2
+        test_data[np.array(position_idx).astype(int) - 1, np.array(time_idx).astype(int)] = 1
```

Comment:
indeed here we don't technically need the kymo that matches the track coordinates, but it's easy enough to have it anyway, might be good to have if this test expands in the future.

That being said, we should probably spend some time in the near future cleaning up and centralizing our test data generation because it is getting a bit unmanageable ... but that's for another day